### ISSUE TITLE
Cannot regenerate the same output

### ISSUE THREAD

**Issue by 14H034160212** [SapienzaNLP/spring#13]
Hi,

I am doing Text-to-AMR task, the input is a text and output is a AMR. the beam size that I used is 1. But when I regenerate twice, the output AMR is not same. How can I make the output AMR be the same for each generation? Like Can I set something like seed to control that? Many thanks.

**Comment by 14H034160212** [comment]
I got the same regeneration after I add the following code before the main function.
```
def setup_seed(seed):
    torch.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
    torch.backends.cudnn.deterministic = True

setup_seed(42)
```

**Comment by mbevila** [comment]
Hi this means you are not setting the model in evaluation mode (model.eval()), with turns off things like dropout, which causes you generate different output.
